service/history/mutableStateBuilder.go

```diff
@@ -2208,6 +2208,11 @@ func (e *mutableStateBuilder) AddContinueAsNewEvent(decisionCompletedEventID int
 		}
 	}
 
+	if useEventsV2 {
```

Comment:
Can you move this logic down to ReplicateWorkflowExecutionContinueAsNewEvent?

Reply:
ok